File: Examples/VisualBasic/Programming-Documents/Joining-Appending/AppendDocumentManually.vb

Imports Microsoft.VisualBasic
Imports System
Imports System.Reflection
Imports System.Collections
Imports System.IO
Imports System.Text

Imports Aspose.Words.Lists
Imports Aspose.Words.Fields
Imports Aspose.Words

Public Class AppendDocumentManually
    Public Shared Sub Run()
        ' ExStart:AppendDocumentManually
        ' The path to the documents directory.
        Dim dataDir As String = RunExamples.GetDataDir_JoiningAndAppending()
        Dim fileName As String = "TestFile.Destination.doc"

        Dim dstDoc As New Document(dataDir & fileName)
        Dim srcDoc As New Document(dataDir & "TestFile.Source.doc")


        Dim mode As ImportFormatMode = ImportFormatMode.KeepSourceFormatting

        ' Loop through all sections in the source document. 
        ' Section nodes are immediate children of the Document node so we can just enumerate the Document.
        For Each srcSection As Section In srcDoc
            ' Because we are copying a section from one document to another, 
            ' it is required to import the Section node into the destination document.
            ' This adjusts any document-specific references to styles, lists, etc.
            '
            ' Importing a node creates a copy of the original node, but the copy
            ' is ready to be inserted into the destination document.
            Dim dstSection As Node = dstDoc.ImportNode(srcSection, True, mode)

            ' Now the new section node can be appended to the destination document.
            dstDoc.AppendChild(dstSection)
        Next

        dataDir = dataDir & RunExamples.GetOutputFilePath(fileName)
        ' Save the joined document
        dstDoc.Save(dataDir)
        ' ExEnd:AppendDocumentManually
        Console.WriteLine(vbNewLine & "Document appended successfully with updated page layout." & vbNewLine & "File saved at " + dataDir)
    End Sub
End Class

File: Examples/CSharp/Programming-Documents/Joining-Appending/AppendDocumentManually.cs
using System;
using System.IO;

using Aspose.Words;

namespace Aspose.Words.Examples.CSharp.Programming_Documents.Joining_and_Appending
{
    class AppendDocumentManually
    {
        public static void Run()
        {
            //ExStart:AppendDocumentManually
            // The path to the documents directory.
            string dataDir = RunExamples.GetDataDir_JoiningAndAppending();
            string fileName = "TestFile.Destination.doc";
            Document dstDoc = new Document(dataDir + fileName);
            Document srcDoc = new Document(dataDir + "TestFile.Source.doc");
            ImportFormatMode mode = ImportFormatMode.KeepSourceFormatting;

            // Loop through all sections in the source document. 
            // Section nodes are immediate children of the Document node so we can just enumerate the Document.
            foreach (Section srcSection in srcDoc)
            {
                // Because we are copying a section from one document to another, 
                // it is required to import the Section node into the destination document.
                // This adjusts any document-specific references to styles, lists, etc.
                //
                // Importing a node creates a copy of the original node, but the copy
                // is ready to be inserted into the destination document.
                Node dstSection = dstDoc.ImportNode(srcSection, true, mode);

                // Now the new section node can be appended to the destination document.
                dstDoc.AppendChild(dstSection);
            }

            dataDir = dataDir + RunExamples.GetOutputFilePath(fileName);
            // Save the joined document
            dstDoc.Save(dataDir);
            //ExEnd:AppendDocumentManually
            Console.WriteLine("\nDocument appended successfully with manual append operation.\nFile saved at " + dataDir);
        }
    }
}
